storm-core/src/jvm/org/apache/storm/container/cgroup/CgroupCenter.java

```diff
@@ -0,0 +1,238 @@
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class CgroupCenter implements CgroupOperation {
+
+    public static Logger LOG = LoggerFactory.getLogger(CgroupCenter.class);
+
+    private static CgroupCenter instance;
+
+    private CgroupCenter() {
+
+    }
+
+    /**
+     * Thread unsafe
+     * 
+     * @return
+     */
+    public synchronized static CgroupCenter getInstance() {
+        if (instance == null)
+            instance = new CgroupCenter();
+        return CgroupUtils.enabled() ? instance : null;
+    }
+
+    @Override
+    public List<Hierarchy> getHierarchies() {
+
+        Map<String, Hierarchy> hierarchies = new HashMap<String, Hierarchy>();
+        FileReader reader = null;
+        BufferedReader br = null;
+        try {
+            reader = new FileReader(Constants.MOUNT_STATUS_FILE);
+            br = new BufferedReader(reader);
+            String str = null;
+            while ((str = br.readLine()) != null) {
+                String[] strSplit = str.split(" ");
+                if (!strSplit[2].equals("cgroup"))
+                    continue;
+                String name = strSplit[0];
+                String type = strSplit[3];
+                String dir = strSplit[1];
+                Hierarchy h = hierarchies.get(type);
+                h = new Hierarchy(name, CgroupUtils.analyse(type), dir);
+                hierarchies.put(type, h);
+            }
+            return new ArrayList<Hierarchy>(hierarchies.values());
+        } catch (Exception e) {
+            LOG.error("Get hierarchies error", e);
+        } finally {
+            CgroupUtils.close(reader, br);
+        }
+        return null;
+    }
+
+    @Override
+    public Set<SubSystem> getSubSystems() {
+
+        Set<SubSystem> subSystems = new HashSet<SubSystem>();
+        FileReader reader = null;
+        BufferedReader br = null;
+        try {
+            reader = new FileReader(Constants.CGROUP_STATUS_FILE);
+            br = new BufferedReader(reader);
+            String str = null;
+            while ((str = br.readLine()) != null) {
+                String[] split = str.split("\t");
+                SubSystemType type = SubSystemType.getSubSystem(split[0]);
+                if (type == null)
+                    continue;
+                subSystems.add(new SubSystem(type, Integer.valueOf(split[1]), Integer.valueOf(split[2]), Integer.valueOf(split[3]).intValue() == 1 ? true
+                        : false));
+            }
+            return subSystems;
+        } catch (Exception e) {
+            LOG.error("Get subSystems error ", e);
+        } finally {
+            CgroupUtils.close(reader, br);
+        }
+        return null;
+    }
+
+    @Override
+    public boolean enabled(SubSystemType subsystem) {
+
+        Set<SubSystem> subSystems = this.getSubSystems();
+        for (SubSystem subSystem : subSystems) {
+            if (subSystem.getType() == subsystem)
```

Comment:
Can you please update all of the blocks to include `{` and `}` even for single line blocks.